htlcswitch/switch.go

```diff
@@ -561,16 +561,86 @@ func (s *Switch) CleanStore(keepPids map[uint64]struct{}) error {
 	return s.attemptStore.CleanStore(keepPids)
 }
 
-// SendHTLC is used by other subsystems which aren't belong to htlc switch
-// package in order to send the htlc update. The attemptID used MUST be unique
-// for this HTLC, and MUST be used only once, otherwise the switch might reject
-// it.
+// SendHTLC is the idempotent, safe entry point for local subsystems to dispatch
+// a payment attempt. If a dispatch error occurs, the attempt is transitioned
+// to a terminal FAILED state before returning the error. This prevents the
+// attempt from being left pending and clarifies it is not in-flight. A return
+// of ErrDuplicateAdd indicates the dispatch for this attempt was already
+// acknowledged. The attempt may be in-flight, or it may already have a terminal
+// (settle or fail) result. The caller MUST use GetAttemptResult to determine
+// the definitive outcome. This allows callers to reliably resume their payment
+// lifecycle after ambiguous failures, such as a restart or network timeout.
 func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, attemptID uint64,
 	htlc *lnwire.UpdateAddHTLC) error {
 
-	// Generate and send new update packet, if error will be received on
-	// this stage it means that packet haven't left boundaries of our
-	// system and something wrong happened.
+	// First, we initialize the attempt in our persistent store. This serves
+	// as a durable record of our intent to send and gates the attempt id
+	// for concurrent callers, allowing them to safely retry.
+	err := s.attemptStore.InitAttempt(attemptID)
```

Comment:
We should also mention that this must always be the first call in this method to not break idempotency

Reply:
added in mention of this. let me know if you like it.